cpp/src/gcylon/sorting/deserialize.cpp

```diff
@@ -0,0 +1,221 @@
+                                                                 buffer_sizes_per_table.at(i));
+        received_tables.push_back(std::move(outTable));
+    }
+
+    return cylon::Status::OK();
+}
+
+std::unique_ptr<cudf::column> constructColumn(std::shared_ptr<rmm::device_buffer> data_buffer,
+                                              std::shared_ptr<rmm::device_buffer> mask_buffer,
+                                              std::shared_ptr<rmm::device_buffer> offsets_buffer,
+                                              cudf::data_type dt,
+                                              int32_t num_rows) {
+
+    std::unique_ptr<cudf::column> clmn = nullptr;
+
+    // if it is non-string column
+    if (dt.id() != cudf::type_id::STRING) {
+        if (mask_buffer->size() == 0) {
+            clmn = std::make_unique<cudf::column>(dt, num_rows, std::move(*data_buffer));
+        } else {
+            clmn = std::make_unique<cudf::column>(dt,
+                                                  num_rows,
+                                                  std::move(*data_buffer),
+                                                  std::move(*mask_buffer));
+        }
+        // if it is a string column
+    } else {
+        // construct chars child column
+        auto cdt = cudf::data_type{cudf::type_id::INT8};
+        auto chars_column = std::make_unique<cudf::column>(cdt, data_buffer->size(), std::move(*data_buffer));
+
+        auto odt = cudf::data_type{cudf::type_id::INT32};
+        auto offsets_column = std::make_unique<cudf::column>(odt, num_rows + 1, std::move(*offsets_buffer));
+
+        std::vector<std::unique_ptr<cudf::column>> children;
+        children.emplace_back(std::move(offsets_column));
+        children.emplace_back(std::move(chars_column));
+
+        if (mask_buffer->size() > 0) {
+            clmn = std::make_unique<cudf::column>(cudf::data_type{cudf::type_id::STRING},
+                                                  num_rows,
+                                                  std::move(rmm::device_buffer{0, rmm::cuda_stream_default}),
+                                                  std::move(*mask_buffer),
+                                                  cudf::UNKNOWN_NULL_COUNT,
+                                                  std::move(children));
+        } else{
+            clmn = std::make_unique<cudf::column>(cudf::data_type{cudf::type_id::STRING},
+                                                  num_rows,
+                                                  std::move(rmm::device_buffer{0, rmm::cuda_stream_default}),
+                                                  std::move(rmm::device_buffer{0, rmm::cuda_stream_default}),
+                                                  0,
+                                                  std::move(children));
+        }
+    }
+    return clmn;
+}
+
+int32_t gcylon::net::numOfRows(const cudf::data_type dt,
```

Comment:
shouldnt num_rows be a int64 value?

Reply:
in cudf, they are using int32-t

Reply:
shall we make this int64? otherwise in arrow stuff we would have to do an explicit downcast?